core/ecschema-metadata/src/Metadata/Class.ts

```diff
@@ -49,6 +48,37 @@ export abstract class ECClass extends SchemaItem implements CustomAttributeConta
       this._modifier = ECClassModifier.None;
   }
 
+  /**
+   * Gets the base class if it exists, otherwise returns undefined.
+   */
+  public get baseClass(): LazyLoadedECClass | undefined {
+    return this._baseClass;
+  }
+
+  /**
+   * Sets the base class of the ECClass. Pass undefined to 'remove' the base class.
+   */
+  public async setBaseClass(baseClass: LazyLoadedECClass | undefined) {
```

Comment:
Do we have to use a LazyLoaded class here? In all calls I saw, we've always had the baseClass resolved already, so there wouldn't be a need to do the detour via the delayed loaded proxies. Even when loading from JSON we expect the baseClass to be in the schema already (lookupItemSync in line 458).

That makes we wonder if we either missed a case where the baseClass is not yet present in the schema (aka lookup would fail) or if having the base class lazy-loaded is over-complicating things there as we have the items resolved already.

Reply:
I think this should be internal and protected. Let me think on this. The editing API should call this. The old property setter  took the LazyLoaded object, and I don't think we should change it.

Reply:
FYI, I updated the method to protected. I know we discussed this on a call a while ago and decided to keep the LazyLoaded stuff for now.